apps/ryal_core/lib/ryal/web/commands/payment_gateway_command.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Ryal.PaymentGatewayCommand do
+  @moduledoc """
+  CRUD commands for `Ryal.PaymentGateway`s. Often used to managed the
+  `external_id`s from payment gateways.
+  """
+
+  alias Ryal.PaymentGateway
+  alias Ryal.PaymentGatewayQuery
+  alias Ryal.PaymentGateway.Customer
+
+  @default_gateway Application.get_env(:ryal_core, :default_payment_gateway)
+  @fallback_gateways Application.get_env(:ryal_core, :fallback_payment_gateways)
+
+  def create(user) do
+    Enum.each @fallback_gateways || [], fn(gateway_type) ->
+      spawn_monitor fn -> create(gateway_type, user) end
+    end
+
+    create @default_gateway, user
+  end
+
+  @doc """
+  Given a user and a gateway type, we'll create a new `Ryal.PaymentGateway` for
+  that user.
+  """
+  @spec create(atom, Ecto.Schema.t) ::
+    {:ok, Ecto.Schema.t} | {:error, Ecto.Changeset.t}
+  def create(type, user) do
+    struct = %PaymentGateway{type: Atom.to_string(type), user_id: user.id}
+
+    with {:ok, external_id} <- Customer.create(type, user),
+         changeset <-
+           PaymentGateway.changeset(%{struct | external_id: external_id}),
+      do: Ryal.repo.insert(changeset)
+  end
+
+  @doc """
+  Give us a user id and we'll update the payment gateways with the user's
+  information.
+  """
+  @spec update(Ecto.Schema.t) :: []
+  def update(user), do: update_payment_gateways(user, :update)
+
+  @doc """
+  If you're going to be deleting your user, then we'll delete the user on the
+  payment gateway as well.
+  """
+  @spec delete(Ecto.Schema.t) :: []
+  def delete(user), do: update_payment_gateways(user, :delete)
+
+  defp update_payment_gateways(user, action) do
+    payment_gateways = user.id
+      |> PaymentGatewayQuery.by_user_id
+      |> Ryal.repo.all
+      |> Ryal.repo.preload(:user)
```

Comment:
I don't like this. I'm given a user, I find its payment gateways, then I preload the user. I should really just preload payment gateways for a user and then call `user.payment_gateways`.